org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -306,9 +327,7 @@ protected void incrementalBuild(IResourceDelta delta, final IProgressMonitor mon
 		final SubMonitor progress = SubMonitor.convert(monitor, Messages.XtextBuilder_CollectingResources, 10);
 		progress.subTask(Messages.XtextBuilder_CollectingResources);
 		
-		if (queuedBuildData.needRebuild(getProject())) {
-			needRebuild();
-		}
+		pollQueuedBuildData();
```

Comment:
@szarnekow this leads to a full build immediately following a project.build(clean)
is this intended?

Reply:
we have deltas for projects that are no related at
all.
but still the full build gets called for the project we called clean on

Reply:
the problematic call is the one in clean.
unfortunately 
the taskqueue ist nailed down so hard to exchange.
will try with subclassing

Reply:
I can't follow yet.
There is a clean on some unrelated project. And then you see a full build here?